frontend/src/hooks/pims-api/useApi.ts

```diff
@@ -1,13 +1,23 @@
-import React from 'react';
+import useDeepCompareMemo from 'hooks/useDeepCompareMemo';
 import { ENVIRONMENT } from 'constants/environment';
-import CustomAxios from 'customAxios';
+import CustomAxios, { defaultEnvelope, LifecycleToasts } from 'customAxios';
 
 /**
  * Common hook to make requests to the PIMS APi.
  * @returns CustomAxios object setup for the PIMS API.
  */
-export const useApi = () => {
-  return React.useMemo(() => CustomAxios({ baseURL: ENVIRONMENT.apiUrl }), []);
+export const useApi = (
+  axiosOptions: {
+    lifecycleToasts?: LifecycleToasts;
+    selector?: Function;
+    envelope?: typeof defaultEnvelope;
+    baseURL?: string;
+  } = {},
+) => {
+  return useDeepCompareMemo(
```

Comment:
Can you explain why this is needed?  Is it because we need a different memo for each axios options?

Reply:
axiosOptions is an object, which will cause a render loop. Either need to make sure all callers only pass memoized objects or use value comparison here (instead of reference comparison).